ApplicationLibCode/ProjectDataModel/WellLog/RimEnsembleWellLogStatistics.cpp

```diff
@@ -127,6 +156,196 @@ void RimEnsembleWellLogStatistics::calculate( const std::vector<RimWellLogFile*>
     }
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+void RimEnsembleWellLogStatistics::calculateByKLayer( const std::vector<RimWellLogFile*>& wellLogFiles,
+                                                      const QString&                      wellLogChannelName )
+{
+    cvf::ref<RigWellLogIndexDepthOffset> offsets = RimEnsembleWellLogStatistics::calculateIndexDepthOffset( wellLogFiles );
```

Comment:
Try to avoid a cvf::object for this data structure. I do not think it is required here. Use std structures if possible.